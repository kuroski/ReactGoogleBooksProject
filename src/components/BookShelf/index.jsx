import React, {Component} from 'react'

const propTypes = {
  books: React.PropTypes.array.isRequired
}

class BookShelf extends Component {
  render() {
    return this.props.books.length ? (
      <ul className="BookShelf">
        {this.props.books.map((item, index) => {
          return <li key={index}>{item.volumeInfo.title}</li>
        })}
      </ul>
    ) : null
  }
}

BookShelf.propTypes = propTypes

export default BookShelf
