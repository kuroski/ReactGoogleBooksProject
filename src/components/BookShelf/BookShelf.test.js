import React from 'react'
import BookShelf from './'
import renderer from 'react-test-renderer'
import {shallow} from 'enzyme'
import booksJson from '../../test/__mocks__/books.json'

describe('BookShelf', () => {
  it('renders correctly', () => {
    const BookShelfTree = renderer.create(
      <BookShelf books={[]} />
    ).toJSON()
    expect(BookShelfTree).toMatchSnapshot()
  })

  it('should render zero books', () => {
    const wrapper = shallow(<BookShelf books={[]} />)
    expect(wrapper.find('li')).toHaveLength(0)
  })

  it('should render some books', () => {
    const booksItems = booksJson.items
    const wrapper = shallow(<BookShelf books={booksItems} />)
    expect(wrapper.find('li')).toHaveLength(2)
  })
})