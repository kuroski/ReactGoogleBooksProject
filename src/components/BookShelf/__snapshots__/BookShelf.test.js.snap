exports[`BookShelf renders correctly 1`] = `null`;
